### PR TITLE
protocol: add test metadata RPC for setting iteam mode

### PR DESCRIPTION
**Pull request by strib**
Issue: KBFS-2869

### CODE REVIEW

File: go/protocol/keybase1/metadata.go [hunk 1]
```diff
@@ -271,6 +271,10 @@ type GetMerkleNodeArg struct {
 	Hash string `codec:"hash" json:"hash"`
 }
 
+type SetImplicitTeamModeForTestArg struct {
+	ImplicitTeamMode string `codec:"implicitTeamMode" json:"implicitTeamMode"`
```
Comment (jzila) [inline go/protocol/keybase1/metadata.go]: Should the mode be an enum instead? Not a big deal obviously since it's just for testing.

Comment (strib) [reply go/protocol/keybase1/metadata.go]: I was just trying to match it with the environment variable behavior, so we wouldn't get confused when comparing the tests to the dockerfile.  But if you think it'd be better as a numeric enum I don't mind changing it.

Comment (strib) [reply go/protocol/keybase1/metadata.go]: (I think this way makes the mdserver code a little cleaner, as I don't have to reconcile two separate enums together.)

Comment (jzila) [reply go/protocol/keybase1/metadata.go]: Yeah fair enough. Can just leave it as a string since it won't ever be called in production clients.

**Pull request merged**